src/treq/client.py

```diff
@@ -147,14 +151,20 @@ def request(self, method, url, **kwargs):
         """
         method = method.encode('ascii').upper()
 
+        if isinstance(url, unicode):
+            parsed_url = URL.from_text(url)
+        else:
+            parsed_url = URL.from_text(url.decode('ascii'))
```

Comment:
Is `ascii` really the right choice here?  This method's supposed to return a `Deferred`, so this creates an awkward error-handling situation at the very least.  I'd probably just go with `charmap` personally, but if you want a `UnicodeDecodeError` maybe make it asynchronous?

Reply:
Apart from the Deferred bit (which seems a defect orthogonal to the one I'm addressing here), I don't really understand the objection here. Could you file a fresh issue with a use case?